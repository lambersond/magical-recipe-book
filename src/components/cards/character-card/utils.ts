import clsx from 'classnames'

export function characterCardStyles(details: 'lite' | 'full') {
  const cardClasses = clsx(
    {
      'gap-0 border-border bg-page/40': details === 'lite',
      'gap-6 border-transparent': details === 'full',
    },
    'border flex flex-col hover:border-primary/50 transition-all duration-200 ease-in-out after:text-text-primary after:absolute after:bottom-4 after:right-4 after:text-xs after:font-thin hover:after:content-["Click_to_view_â†’"] hover:after:text-primary relative',
  )
  const headerClasses = clsx(
    {
      'gap-2': details === 'lite',
      'gap-4': details === 'full',
    },
    'flex',
  )
  const nameClasses = clsx(
    {
      'text-xl lg:text-2xl -mt-1.25': details === 'lite',
      'text-3xl lg:text-5xl': details === 'full',
    },
    'font-bold truncate bg-gradient-to-r from-primary to-secondary bg-clip-text text-transparent text-left',
  )
  const dayClasses = clsx(
    {
      'text-xs lg:text-md -mt-2.5': details === 'lite',
      'text-sm lg:text-lg': details === 'full',
    },
    'text-text-secondary flex gap-1 items-baseline',
  )
  const numberClasses = clsx(
    {
      'text-xl': details === 'lite',
      'text-2xl': details === 'full',
    },
    'text-text-primary font-bold',
  )
  const statsContainerClasses = clsx(
    {
      'grid-cols-1 md:grid-cols-3': details === 'full',
      'grid-cols-3': details === 'lite',
    },
    'grid gap-3 mb-auto',
  )
  const headerTextClasses = clsx(
    {
      'h-fit': details === 'lite',
      'h-unset': details === 'full',
    },
    'grid gap-1',
  )

  return {
    cardClasses,
    headerClasses,
    nameClasses,
    dayClasses,
    numberClasses,
    statsContainerClasses,
    headerTextClasses,
  }
}

export function characterCardStatStyles(details: 'lite' | 'full') {
  const valueClasses = clsx(
    {
      'text-lg': details === 'lite',
      'text-5xl': details === 'full',
    },
    'font-bold text-text-primary',
  )

  const containerClasses = clsx(
    {
      'gap-2 py-2': details === 'lite',
      'bg-page/50 p-4 flex-col items-center justify-center': details === 'full',
    },
    'rounded-lg flex',
  )

  return { valueClasses, containerClasses }
}
