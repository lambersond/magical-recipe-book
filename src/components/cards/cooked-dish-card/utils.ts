import clsx from 'classnames'
import type { CookedDishRecipe, CookedDishStatus } from '@/types'

export function bonusesListClasses(isPrepared: boolean, type: 'boon' | 'bane') {
  const containerClasses = clsx('border-l-2 pl-2 py-2 rounded-l-lg', {
    'border-success bg-success/10': type === 'boon',
    'border-danger bg-danger/10': type === 'bane',
  })

  const titleClasses = clsx(
    {
      'text-success/80': type === 'boon',
      'text-danger/80': type === 'bane',
      'before:content-["Potential"] flex gap-1': isPrepared,
    },
    'font-semibold capitalize',
  )

  const itemClasses = clsx(
    {
      'text-success/80': type === 'boon',
      'text-danger/80': type === 'bane',
    },
    'text-sm',
  )

  return {
    containerClasses,
    itemClasses,
    titleClasses,
  }
}

export function deriveStatusChipDetails(
  isExpired: boolean,
  isPrepared: boolean,
  cookedOnDay: number,
  consumedOnDay: number | null,
  currentDay: number,
) {
  let text = ''
  let value = ''
  let chipClasses = ''

  if (isExpired) {
    text = 'Spoiled on Day'
    value = (cookedOnDay + 7).toString()
    chipClasses = 'before:content-["üíÄ"] bg-danger/10 text-danger'
  } else if (isPrepared) {
    const daysRemaining = 7 - (currentDay - cookedOnDay)
    text = 'Expires '
    value =
      daysRemaining === 0
        ? 'Today!'
        : `in ${daysRemaining} day${daysRemaining === 1 ? '' : 's'}`
    chipClasses = clsx('before:content-["‚è∞"]', {
      'bg-danger/10 text-danger': daysRemaining <= 1,
      'bg-warning/10 text-warning': daysRemaining > 1 && daysRemaining <= 3,
      'bg-success/10 text-success': daysRemaining > 3,
    })
  } else if (consumedOnDay) {
    text = 'Consumed on Day'
    value = consumedOnDay.toString()
    chipClasses = 'before:content-["üç¥"] bg-info/10 text-info'
  }

  const containerClasses = clsx(
    'px-3 py-1 rounded-full text-sm flex items-center before:mr-1',
    chipClasses,
  )

  return {
    text,
    value,
    containerClasses,
  }
}

export function actionButtonClasses(type: 'consume' | 'discard') {
  return clsx(
    'px-3 py-1 text-lg font-semibold rounded items-center transition-colors flex items-center gap-1 cursor-pointer',
    {
      'bg-success/80 hover:bg-success text-text-primary md:after:content-["Consume"]':
        type === 'consume',
      'bg-danger/80 hover:bg-danger text-text-primary md:after:content-["Discard"]':
        type === 'discard',
    },
  )
}

export function deriveCookedDishCardProperties(
  currentDay: number,
  cookedOnDay: number,
  status: CookedDishStatus,
) {
  const isPrepared = status === 'prepared'
  const isExpired = currentDay > cookedOnDay + 7

  return {
    isPrepared,
    isExpired,
  }
}

export function getDescriptionText(
  recipe: CookedDishRecipe,
  status: CookedDishStatus,
) {
  if (['boon', 'bane'].includes(status)) {
    return recipe[`${status as 'boon' | 'bane'}Text`]
  }

  return recipe.description
}

export function getCookedStatusClasses(status: CookedDishStatus) {
  return clsx(
    {
      'before:content-["üçΩÔ∏è"]': status === 'prepared',
      'before:content-["‚ú®"]': status === 'boon',
      'before:content-["ü§§"]': status === 'success',
      'before:content-["ü§¢"]': status === 'failure',
      'before:content-["üíÄ"]': status === 'bane',
    },
    'flex gap-1 bg-gray-700 text-text-secondary text-sm rounded-full py-1 px-3 capitalize',
  )
}
