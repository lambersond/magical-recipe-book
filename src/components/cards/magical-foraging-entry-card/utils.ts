import clsx from 'classnames'
import type { MagicalForagingEntryCardProps } from './types'

export function getStyles({
  rarity,
  isExpired,
  isUsed,
}: Pick<MagicalForagingEntryCardProps, 'rarity' | 'isExpired' | 'isUsed'>) {
  const cardClasses = clsx(
    'max-w-xl py-3 rounded-xl px-2',
    'border border-border bg-card',
  )
  const itemClasses = clsx('text-xl font-bold', `text-rarity-${rarity}`)
  const statusClasses = clsx('text-xs', 'font-semibold', 'text-text-secondary')
  const statusIndicatorClasses = clsx(
    {
      'bg-success': !isExpired && !isUsed,
      'bg-error': isExpired,
      'bg-info': isUsed,
    },
    'size-2 rounded-full',
  )

  return {
    cardClasses,
    itemClasses,
    statusClasses,
    statusIndicatorClasses,
  }
}

export function getStatusText({
  isExpired,
  isUsed,
}: Pick<MagicalForagingEntryCardProps, 'isExpired' | 'isUsed'>) {
  if (isUsed) {
    return 'Used'
  } else if (isExpired) {
    return 'Expired'
  }
  return 'Available'
}
