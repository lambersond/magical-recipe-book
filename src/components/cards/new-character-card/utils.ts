import clsx from 'classnames'
import type { NewCharacterCardProps } from './types'

export function newCharacterCardStyles({
  color,
  variant,
}: Required<NewCharacterCardProps>) {
  const containerClasses = clsx(
    {
      'text-text-primary': color === 'primary',
      'text-text-secondary hover:text-text-primary': color === 'secondary',
    },
    'flex flex-col items-center justify-center border-dashed border-2 border-text-primary/30 cursor-pointer hover:border-text-primary',
  )

  const titleClasses = clsx(
    {
      'text-lg before:content-["+"] before:mr-2': variant === 'small',
      'text-2xl mb-2': variant === 'large',
    },
    'font-bold',
  )

  return {
    containerClasses,
    titleClasses,
  }
}
