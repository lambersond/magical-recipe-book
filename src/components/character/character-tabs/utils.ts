export const tabStyles =
  'flex items-center gap-2 text-sm transition-colors border-b-2 py-3 border-b-transparent hover:border-b-primary text-text-secondary hover:text-text-primary focus:outline-none'
export const tabPanelStyles = 'max-w-7xl mx-auto px-0 sm:px-6 lg:px-8 w-full'
