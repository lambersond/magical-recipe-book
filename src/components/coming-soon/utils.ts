export const getStatusText = (status: string) => {
  switch (status) {
    case 'completed': {
      return 'Completed'
    }
    case 'in-progress': {
      return 'In Progress'
    }
    default: {
      return 'Coming Soon'
    }
  }
}

export const getStatusColor = (status: string) => {
  switch (status) {
    case 'completed': {
      return 'border-green-500/30 bg-green-500/10'
    }
    case 'in-progress': {
      return 'border-yellow-500/30 bg-yellow-500/10'
    }
    default: {
      return 'border-gray-500/30 bg-gray-500/10'
    }
  }
}
