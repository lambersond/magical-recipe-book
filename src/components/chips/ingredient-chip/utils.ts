import { getRarityContainerClasses } from '@/utils/rarity'
import type { Rarity } from '@/types'

export function getStyles(rarity: Rarity) {
  const containerClasses = getRarityContainerClasses(
    rarity,
    'text-xs rounded-sm px-2 py-1 border flex min-w-fit items-center',
  )

  return {
    containerClasses,
  }
}
