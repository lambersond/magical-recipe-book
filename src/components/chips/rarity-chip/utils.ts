import clsx from 'classnames'
import { getRarityContainerClasses, getRarityTextColor } from '@/utils/rarity'
import type { Rarity } from '@/types'

export function getStyles(rarity: Rarity, iconOnly?: boolean) {
  const containerClasses = getRarityContainerClasses(
    rarity,
    'text-xs rounded-full px-2 py-1 border flex min-w-fit items-center',
  )
  const iconClasses = clsx(
    {
      'sm:mr-2': !iconOnly,
    },
    'inline size-4',
    getRarityTextColor(rarity, ''),
  )

  return {
    containerClasses,
    iconClasses,
  }
}
