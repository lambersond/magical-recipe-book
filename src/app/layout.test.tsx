import { render } from '@testing-library/react'
import RootLayout from './layout'
import { DOM_NESTING_ERROR } from '@/utils/test-utils'

jest.mock('@/components/session-wrapper', () => ({
  SessionWrapper: ({ children }: { children: React.ReactNode }) => (
    <div>{children}</div>
  ),
}))

describe('app/layout', () => {
  it('should render layout', () => {
    jest.spyOn(console, 'error').mockImplementation((msg: string) => {
      if (msg.includes(DOM_NESTING_ERROR)) return
      console.error(msg)
    })

    const { getByText, asFragment } = render(
      <RootLayout>Hello World</RootLayout>,
    )
    expect(getByText('Hello World')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
