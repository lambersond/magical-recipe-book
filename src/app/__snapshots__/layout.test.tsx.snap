// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`app/layout should render layout 1`] = `
<DocumentFragment>
  <div>
    Hello World
  </div>
</DocumentFragment>
`;
